core/consensus/common/src/block_import.rs

```diff
@@ -137,6 +137,7 @@ pub struct BlockImportParams<Block: BlockT> {
 
 impl<Block: BlockT> BlockImportParams<Block> {
 	/// Deconstruct the justified header into parts.
+	#[allow(clippy::type_complexity)]
```

Comment:
Please remove

Reply:
These can absolutely be removed, but then the issue is that `clippy` will not ignore these, and their cause cannot be fixed - in this case, such a refactoring would be far beyond the scope of this PR.

If these flags are not acceptable to you, I'm ok with just considering this PR as is (without them), and leaving the addition of `clippy` to CI to another contribution.